plugins/pulp_docker/plugins/registry.py

```diff
@@ -529,6 +539,35 @@ def _get_path(self, path, headers=None):
 
         return report.headers, report.destination.getvalue()
 
+    def _get_path(self, path, headers=None):
+        """
+        Retrieve a single path within the upstream registry, and return a 2-tuple of the headers and
+        the response body.
+
+        :param path: a full http path to retrieve that will be urljoin'd to the upstream registry
+                     url.
+        :type  path: basestring
+        :param headers: headers sent in the request
+        :type headers:  dict
+
+        :return:     (headers, response body)
+        :rtype:      tuple
+        """
+        backoff = 1  # 2s initial backoff
+        while True:
+            try:
+                return self._get_one_path(path)
+            except TooManyRequestsError as e:
+                if e.retry_after is None:
+                    # No retry-after header provided, use exponential backoff
+                    if backoff < 2**11:  # Maximum backoff
```

Comment:
what is the reason for such high backoff limit? can we decrease it? celery's default max backoff is 600 seconds.

Reply:
I'll bring it down to 2**9 (512s).